db/migrate/20150228200007_split_annotations.rb

```diff
@@ -0,0 +1,43 @@
+class SplitAnnotations < ActiveRecord::Migration
+  def change
+
+    Annotation.find_each do |annotation|
+        text = annotation.text
+
+
+        res = text.split("[");
+        
+        if res.size == 1 then
+            annotation.comment = text
+        end
+
+        if res.size > 1 then
+
+            len = res.size
+            points = res[len-1]
+            res.delete_at(len-1)
+
+            points = points.split(":");
+            comment = res.join("[");
+            annotation.comment = comment
+           
+            if points.size == 1 then
+              annotation.value = points[0].delete("]").to_f
+            else
+              if points[0] == "?" then
+                annotation.problem = points[1].delete("]").to_i
```

Comment:
This is an example of after monday: "5sdfzdf[5:problem2]"

I thought problem2 would be the ID, but it might actually be the name of the problem... My b.